x-pack/plugins/osquery/server/routes/privileges_check/privileges_check_route.ts

```diff
@@ -22,21 +21,29 @@ export const privilegesCheckRoute = (router: IRouter, osqueryContext: OsqueryApp
     async (context, request, response) => {
       const esClient = context.core.elasticsearch.client.asCurrentUser;
 
-      const privileges = (
-        await esClient.security.hasPrivileges({
-          body: {
-            index: [
-              {
-                names: [`logs-${OSQUERY_INTEGRATION_NAME}.result*`],
-                privileges: ['read'],
-              },
-            ],
-          },
-        })
-      ).body;
+      if (osqueryContext.security.authz.mode.useRbacForRequest(request)) {
+        const privileges = (
+          await esClient.security.hasPrivileges({
+            body: {
+              index: [
+                {
+                  names: [`logs-${OSQUERY_INTEGRATION_NAME}.result*`],
+                  privileges: ['read'],
+                },
+              ],
+            },
+          })
```

Comment:
Instead of using the ES client directly for this privilege check, you can use the abstraction that the Security plugin provides:

```ts
osqueryContext.security.authz.checkPrivilegesDynamicallyWithRequest({
  elasticsearch: {
    index: [
      {
        names: [`logs-${OSQUERY_INTEGRATION_NAME}.result*`],
        privileges: ['read'],
      },
    ]
  }
})
```

One upside of this is that the Security plugin performs response validation. This is the preferred approach Kibana-wide for performing ES privilege checks.